core/src/lib.rs

```diff
@@ -93,6 +94,17 @@ pub struct CoreInitOptions {
     pub telemetry_opts: TelemetryOptions,
 }
 
+/// TODO:  This holds an actual worker now. Lang keeps track of taskq->worker mapping
+pub struct WorkerImpl {}
```

Comment:
Good chance this just ends up being the actual worker struct

Reply:
:+1: Since it's all internal no need to prematurely abstract to trait